SEED-platform/seed
# BuildingSync Upload

#### What's this PR do?
Adds UI support for uploading BuildingSync files independently as new property records, and UI support for updating existing properties with BuildingSync files.

These changes include improved error handling and error messages for invalid file types, invalid files, incorrect schemas, etc.

**BONUS** - this PR upgraded to the latest version of FineUploader, which means it has ~6 years of improvements, no longer needs a compile step, now supports more than POST methods, and is using the slimmed down Core module instead of the UI module.
#### How should this be manually tested?
- Using the main upload dialog, upload a BuildingSync file and check that it created a new record in the Inventory List view
    - Also try uploading a non-xml file
    - Also try uploading a BuildingSync file with an older, incompatible schema

- Click the (i) icon on any Property record in the Inventory List view to go to the detail page, and from there upload a BuildingSync file.  The page should show upload progress and reload upon success or issue an error.

#### What are the relevant tickets?
#1685 #876
#### Screenshots (if appropriate)

*Upload BuildingSync Data* button
![2018-08-14_154406](https://user-images.githubusercontent.com/411466/44120628-34310036-9fda-11e8-8dce-c66f0f1d3a9b.png)

*BuildingSync Update* button
![2018-08-14_155421](https://user-images.githubusercontent.com/411466/44120687-5deeadec-9fda-11e8-92cd-b4b7a6aa92f7.png)

#### Definition of Done:
